### PR TITLE
Refactor scheduler

### LINKED ISSUE #366: Refactor scheduler task manager
### Feature request:

The TaskManager logic is a bit messy and needs to be cleaned up
### Use case:

pkg.TaskId = s.service.GenerateTaskID(request.Url, request.Filter, request.UrlMeta, request.BizId, request.PeerId)
	task, ok := s.service.GetTask(pkg.TaskId)
	if !ok {
		task, err = s.service.AddTask(&types.Task{
			TaskID:  pkg.TaskId,
			URL:     request.Url,
			Filter:  request.Filter,
			BizID:   request.BizId,
			URLMata: request.UrlMeta,
		})
		if err != nil {
			dferror, _ := err.(*dferrors.DfError)
			if dferror != nil && dferror.Code == dfcodes.SchedNeedBackSource {
				isCdn = true
			} else {
				return
			}
		}
	}

### PR DESCRIPTION
**Pull request by 244372610**
## Related Issue
fix #366 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist:

- [x] Code compiles correctly.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: internal/idgen/uuid.go [hunk 1]
```diff
@@ -16,8 +16,16 @@
...
+
+	"github.com/google/uuid"
+)
 
 func UUIDString() string {
 	return uuid.New().String()
 }
+
+func CDNUUID(cdnHostName string, serverPort int32) string {
+	return fmt.Sprintf("%s-%d_CDN", cdnHostName, serverPort)
```
Comment (jim3ma) [inline internal/idgen/uuid.go:30]: Where is uuid ?

File: scheduler/daemon/gc/manager.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+//
+//func newManager(cfg *config.GCConfig) daemon.GCMgr {
+//	return &manager{
+//		gcInterval: time.Duration(3),
+//		done:       nil,
+//	}
+//}
+
+func (m *manager) StartGC(ctx context.Context) error {
```
Comment (gaius-qi) [inline scheduler/daemon/gc/manager.go:41]: Use `pkg/gc` library.

File: scheduler/daemon/host/host_manager.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/*
```
Comment (gaius-qi) [inline scheduler/daemon/host/host_manager.go:1]: Unified naming scheduler/daemon/host/host_manager.go and scheduler/daemon/host_mgr.go

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: tmp [hunk 4]
```diff
@@ -0,0 +1,283 @@
+<!DOCTYPE html><!--STATUS OK-->
```
Comment (jim3ma) [inline tmp:1]: Unused file.

File: docs/en/config/dfget.yaml [hunk 5]
```diff
@@ -8,12 +8,12 @@ gcInterval: 1m0s
 # daemon data directory
 # in linux, default value is $HOME/.dragonfly/dfget-daemon/
 # in macos(just for testing), default value is /Users/$USER/.dragonfly/dfdaemon/
-dataDir: /var/lib/dragonfly
+dataDir: /Users/sunweipeng1/dragonfly
```
Comment (jim3ma) [inline docs/en/config/dfget.yaml:11]: Reset config.

File: docs/en/config/dfget.yaml [hunk 6]
```diff
@@ -8,12 +8,12 @@ gcInterval: 1m0s
...
 
 # daemon work directory, daemon will change current working directory to this
 # in linux, default value is $HOME/.dragonfly/dfget-daemon/
 # in macos(just for testing), default value is /Users/$USER/.dragonfly/dfdaemon/
-workHome: /var/lib/dragonfly
+workHome: /Users/sunweipeng1/dragonfly
```
Comment (jim3ma) [inline docs/en/config/dfget.yaml:16]: Reset config.

File: pkg/rpc/scheduler/scheduler.proto [hunk 7]
```diff
@@ -127,6 +127,7 @@ message PeerPacket{
   string src_pid = 3;
 
   // concurrent downloading count from main peer
+  // todo what effect
```
Comment (jim3ma) [inline pkg/rpc/scheduler/scheduler.proto:130]: `todo` -> `TODO`

**Review by gaius-qi — APPROVED** [review]
LGTM

**Review by jim3ma — APPROVED** [review]

**Pull request merged**